### ISSUE TITLE
Support lodash style throttling options for trailing+maxWidth

### ISSUE THREAD

**Issue by tryggvigy** [xnimorz/use-debounce#60]
https://github.com/lodash/lodash/blob/master/throttle.js

It would be nice to be able to do this:

```ts
useDebouncedCallback(callback, 300, {
  leading: true,
  trailing: false,
  maxWait: 300,
}),
```

where the trailing edge is turned off.
Let's say the function is called twice in the first 300ms.

**What happened**
Debounced function is called _twice_ even though trailing is false (this is due to the maxWait logic).

**What I would have expected**
Debounced function to have been called _once_.

From [Debouncing and Throttling Explained Through Examples](https://css-tricks.com/debouncing-throttling-explained-examples/#debounce-implementations) which is linked in lodash source:

> The new maxWait option (only in Lodash at the moment) is not covered in this article but it can be very useful. Actually, the throttle function is defined using _.debounce with maxWait, as you see in the lodash source code.

**Comment by tryggvigy** [comment]
I think we'd need to keep track of calls with `Date.now()` and compare agains maxWait to achieve this behaviour.
[This is what lodash debounce does](https://github.com/lodash/lodash/blob/86a852fe763935bb64c12589df5391fd7d3bb14d/debounce.js#L185).

**Comment by xnimorz** [comment]
Hi @tryggvigy 
Good point, you are welcome to open a PR, or I'll make these changes later when I would have free time.

**Comment by tryggvigy** [comment]
fixed in https://github.com/xnimorz/use-debounce/pull/62

**Comment by xnimorz** [comment]
Published use-debounce@4.0.0
